domains/pallets/evm-tracker/src/fees.rs

```diff
@@ -0,0 +1,61 @@
+//! Fees module for EVM domain
+
+use crate::Config;
+use core::marker::PhantomData;
+use domain_runtime_primitives::Balance;
+use pallet_block_fees::Pallet as BlockFees;
+use pallet_evm::FeeCalculator;
+use pallet_transaction_payment::Pallet as TransactionPayment;
+use sp_core::U256;
+use sp_evm_tracker::WEIGHT_PER_GAS;
+use sp_runtime::traits::Get;
+use sp_runtime::{FixedPointNumber, Perbill};
+use sp_weights::Weight;
+
+/// Evm gas price calculator for EVM domains.
+/// TransactionWeightFee is the fee for 1 unit of Weight.
+/// GasPerByte is the gas for 1 byte
+pub struct EvmGasPriceCalculator<T, TransactionWeightFee, GasPerByte, StorageFeePercent>(
+    PhantomData<(T, TransactionWeightFee, GasPerByte, StorageFeePercent)>,
+);
+
+impl<T, TransactionWeightFee, GasPerByte, StorageFeePercent> FeeCalculator
+    for EvmGasPriceCalculator<T, TransactionWeightFee, GasPerByte, StorageFeePercent>
+where
+    T: Config
+        + frame_system::Config
+        + pallet_transaction_payment::Config
+        + pallet_block_fees::Config<Balance = Balance>,
+    TransactionWeightFee: Get<T::Balance>,
+    GasPerByte: Get<T::Balance>,
+{
+    fn min_gas_price() -> (U256, Weight) {
+        // spread the storage fee across the gas price based on the Gas Per Byte.
+        let storage_fee_per_gas =
+            BlockFees::<T>::final_domain_transaction_byte_fee().saturating_div(GasPerByte::get());
```

Comment:
`saturating_div` is meaningless for unsigned integers, because the only division that can overflow a type is `INT_MIN / -1`. (Numerically, that’s one more than INT_MAX.)

Do we want div_ceil() here instead, to round up in the case where the byte fee is less than the gas per byte?
(Or the result is a fraction that’s close to the next whole integer.)
https://doc.rust-lang.org/std/primitive.u128.html#method.div_ceil

This would also be consistent with the `mul_ceil` in the storage fee split below.